app/router.go

```diff
@@ -58,7 +63,28 @@ func Router(c Reporter) *mux.Router {
 	return router
 }
 
-func captureTopology(rep Reporter, f func(Reporter, topologyView, http.ResponseWriter, *http.Request)) func(http.ResponseWriter, *http.Request) {
+type collector interface {
```

Comment:
It was there originally, but I think it makes more sense here, as the concept of an abstract collector is only useful to the app. More generally, because interfaces define expected behavior, they're actually tools for consumers, rather than producers, of APIs. Ben Johnson gets at it in an article on testing:

> [the] caller should create the interface instead of the callee providing an interface. This makes sense because the caller can declare exactly what it needs.

—https://medium.com/@benbjohnson/structuring-tests-in-go-46ddee7a25c